src/TaskLineRenderer.ts

```diff
@@ -53,6 +45,41 @@ export class FieldLayoutDetail {
     public static noAttributeValueCalculator: AttributeValueCalculator = () => {
         return '';
     };
+    public static dateAttributeCalculator = (component: TaskLayoutComponent, task: Task) => {
+        /**
+         * Translate a relative date to a CSS class: 'today', 'future-1d' (for tomorrow), 'past-1d' (for yesterday)
+         * etc.
+         * A cutoff (in days) is defined in MAX_DAY_VALUE_RANGE, from beyond that a generic 'far' postfix will be added.
+         * (the cutoff exists because we don't want to flood the DOM with potentially hundreds of unique classes.)
+         */
+        const MAX_DAY_VALUE_RANGE = 7;
+        const DAY_VALUE_OVER_RANGE_POSTFIX = 'far';
+        function dateToAttribute(date: Moment) {
+            const today = window.moment().startOf('day');
+            let result = '';
+            const diffDays = today.diff(date, 'days');
+            if (isNaN(diffDays)) return null;
+            if (diffDays === 0) return 'today';
+            else if (diffDays > 0) result += 'past-';
+            else if (diffDays < 0) result += 'future-';
+            if (Math.abs(diffDays) <= MAX_DAY_VALUE_RANGE) {
+                result += Math.abs(diffDays).toString() + 'd';
+            } else {
+                result += DAY_VALUE_OVER_RANGE_POSTFIX;
+            }
+            return result;
+        }
+
+        const date = task[component];
+
+        if (date instanceof window.moment) {
+            const attributeValue = dateToAttribute(date);
+            if (attributeValue) {
+                return attributeValue;
+            }
+        }
+        return '';
+    };
```

Comment:
Small point, but this is  more complex than the previous code. We can sort it out later.